src/VecSim/algorithms/hnsw/hnsw.h

```diff
@@ -1829,6 +1829,12 @@ VecSimQueryResult_List HNSWIndex<DataType, DistType>::topKQuery(const void *quer
     idType bottom_layer_ep = searchBottomLayerEP(query_data, timeoutCtx, &rl.code);
     if (VecSim_OK != rl.code) {
         return rl;
+    } else if (bottom_layer_ep == INVALID_ID) {
+        // Although we checked that the index is not empty (cur_element_count == 0), it might be
```

Comment:
do we have a test for this scenario?

Reply:
the test of inserting and searching at the same time catches this case